### PR TITLE
[PCFG-29] counterpart

### PR DESCRIPTION
**Pull request by jacopocarlini**
Implemented GET `/counterparttables`
Implemented GET `/counterparttables/:id`

#### List of Changes

- implemented business logic in `CounterpartService.java`
- added entities for `Counterpart`

#### Motivation and Context

To introduce new features

#### How Has This Been Tested?

Tested with Junit

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -39,3 +39,4 @@ Results/
 **/node_modules
 **/tmp.json
 **/k6-script.js
+/src/main/resources/application-sia.properties
```
Comment (pasqualespica) [inline .gitignore:42]: why ?

File: src/main/java/it/pagopa/pagopa/apiconfig/controller/CounterpartController.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    @GetMapping(
+            value = "/{idcounterparttable}",
+            produces = {"application/xml"}
+    )
+    public ResponseEntity<Resource> getCounterpartTable(
+            @Parameter(description = "Id counterpart table", required = true) @PathVariable("idcounterparttable") String idCounterpartTable,
+            @Parameter(description = "Creditor institution code", required = true) @RequestParam("creditorinstitutioncode") String creditorInstitutionCode) {
+        byte[] file = counterpartService.getCounterpartTable(idCounterpartTable, creditorInstitutionCode);
+        return ResponseEntity.ok()
+                .contentType(MediaType.APPLICATION_XML)
```
Comment (pasqualespica) [inline src/main/java/it/pagopa/pagopa/apiconfig/controller/CounterpartController.java:69]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#important_mime_types_for_web_developers  ?

**Review by pasqualespica — COMMENTED** [review]
I don't see the updated postman collection. is this correct?

**Comment by jacopocarlini** [comment]
> I don't see the updated postman collection. is this correct?

You are right. I have updated the file

**Pull request merged**